app/knight.py

```diff
@@ -0,0 +1,53 @@
+class Knight:
+    def __init__(self, config: dict) -> None:
+        self.name = config["name"]
+        self.power = self.apply_power(config)
+        self.protection = self.apply_armour(config)
+        self.hp = self.apply_hp(config)
+
+    @staticmethod
+    def apply_armour(config) -> int:
+        protection = 0
+
+        if config["potion"] is not None:
+            if "protection" in config["potion"]["effect"]:
+                protection += config["potion"]["effect"]["protection"]
+
+        for a in config["armour"]:
+            protection += a["protection"]
+        return protection
+
+    @staticmethod
+    def apply_power(config) -> int:
+        power = config["power"] + config["weapon"]["power"]
+
+        if config["potion"] is not None:
+            if "power" in config["potion"]["effect"]:
+                power += config["potion"]["effect"]["power"]
```

Comment:
Similar to the apply_armour method, ensure that 'potion' and 'effect' keys exist to avoid KeyErrors.